### ISSUE TITLE
Bullet PR: Body position is changed for one tick on mode change

### ISSUE THREAD

**Issue by kubecz3k** [godotengine/godot#12289]
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
Bullet branch 2152ca7dc73a943fcd10d86b024687c2955bcd14
relase_debug build

**Issue description:**
This issue is similar and probably related to https://github.com/godotengine/godot/issues/12267

When body is changing mode to Kinematic, and some movement is done in kinematic it will be for one game tick on it's old position when changing back to Rigid/Character: 
https://youtu.be/ZiEdh896Jro

**Steps to reproduce:**
1. Create RigidBody
2. Add Tween node named  to it.
3.  set `userImitater` playback mode to to `Physics`
4. Fire up this code during the runtime
```
func imitateMovementInKinematic():
	if(get_mode()!=RigidBody.MODE_KINEMATIC):
		set_mode(RigidBody.MODE_KINEMATIC);
		get_node("../../Label").set_text("Setting CYLINDER to Kinematic")
		userImitater.interpolate_property(self, "translation", translation, translation+Vector3(0,2,0), 0.5, Tween.TRANS_QUAD, Tween.EASE_IN_OUT);
		userImitater.start();
		yield(get_tree().create_timer(0.75), "timeout");
	        set_mode(RigidBody.MODE_RIGID); 
		get_node("../../Label").set_text("Setting CYLINDER to rigid")
```
5. You should be able to see quick body flash when its mode is changed from Kinematic to Character

Quick way:
1. Download and run sample project.
2. The reproduction  logic of it is in `IssueHero.tscn` 

**Link to minimal example project:**
[v2BodyPositionFlashOnModeChange.zip](https://github.com/godotengine/godot/files/1404360/v2BodyPositionFlashOnModeChange.zip)

**Comment by kubecz3k** [comment]
cc @AndreaCatania  I think this one was also introduced by 8fd726d

**Comment by kubecz3k** [comment]
Yeah to be sure I double checked and 8fd726dad0cf9588a6316efa3bbee6e68006f508 is the first bad commit

**Comment by AndreaCatania** [comment]
Fixed ec3ff80

**Comment by kubecz3k** [comment]
it works! :)
